platform/device/get_devices.go

```diff
@@ -48,9 +49,13 @@ type getDevicesResponse struct {
 func (r getDevicesResponse) Failed() error { return r.Err }
 
 func decodeListDevicesRequest(ctx context.Context, r *http.Request) (interface{}, error) {
-	defer r.Body.Close()
 	var opts ListDevicesOption
-	req := getDevicesRequest{Opts: opts}
+	if err := json.NewDecoder(r.Body).Decode(&opts); err != nil {
```

Comment:
I would take a look at using `httputil.DecodeJSONRequest` — does a bunch of handy JSON/error handling for you. Here's an example usage: https://github.com/micromdm/micromdm/blob/e32d2bea1b56aa121abf97953ab16131d7d01e39/platform/dep/get_dep_device.go#L31-L35

Reply:
handy!